numpy/core/src/multiarray/einsum.c.src

```diff
@@ -341,6 +369,25 @@ finish_after_unrolled_loop:
         /* Finish off the loop */
         goto finish_after_unrolled_loop;
     }
+#elif EINSUM_USE_NEON_64 && @float64@
+        /* Unroll the loop by 8 */
+        while (count >= 8) {
```

Comment:
Since in NPYV, we're dealing with different SIMD width, you can replace `8` with `npyv_nlanes_f64 * 4` where npyv_nlanes_{suffix type} is representing
the number of vector lanes.

*Note*: that may lead to a negative impact on avx512 since you're
unrolling a big chunk of memory.

However,  you can just unroll by 2 vectors on avx512, for example:
```C
#if NPY_SIMD_WIDTH > 32 # avx512f
const int unroll_by = npyv_nlanes_f64 * 2;
#else
const int unroll_by = npyv_nlanes_f64 * 4;
#endif
while (count >= unroll_by) {
  // ...
}
```